brd-android/app/src/main/java/com/breadwallet/ui/usdaccount/transactions/UsdAccountTransactionsHandler.kt

```diff
@@ -20,25 +20,42 @@ fun createUsdAccountDetailsHandler(
 ): FlowTransformer<F, E> = subtypeEffectHandler {
 
     addTransformer<F.LoadTransactions> { effects ->
-        // todo: add support for pagination
-        effects.flatMapLatest { flowOf(exchangeApi.getFboTransactions()) }
-            .mapLatest { response ->
+        effects.flatMapLatest { effect ->
+            flowOf(
+                effect.previousTransactions to
+                    exchangeApi.getFboTransactions(limit = effect.limit, offset = effect.offset)
+            )
+        }
+            .mapLatest { pair ->
+                val previousTransactions = pair.first
+                val response = pair.second
+
                 when (response) {
                     is ApiResponse.Success -> {
                         val transactions = response.data?.transactions?.map {
                             it.toTransactionListItem(context)
                         } ?: emptyList()
 
                         val groupedTransactions = transactions
-                            .sortedByDescending { it.date }
                             .groupBy { groupsDateFormat.format(it.date) }
                             .map {
                                 val section = TransactionListItem.Section(it.key)
-                                listOf<TransactionListItem>(section) + it.value
+
+                                if (previousTransactions.contains(section)) {
+                                    it.value
+                                } else {
+                                    listOf<TransactionListItem>(section) + it.value
+                                }
                             }
                             .flatten()
+                            .toMutableList()
+
+                        // todo remove when BE fix ready
+                        val hasNextPage = groupedTransactions
+                            .filterIsInstance<TransactionListItem.Transaction>().size ==
+                            UsdAccountTransactionsController.TRANSACTION_PAGE_LIMIT
```

Comment:
Maybe we should use modus operand here to fetch more than just two pages?

```suggestion
                        val hasNextPage = groupedTransactions
                            .filterIsInstance<TransactionListItem.Transaction>().size % UsdAccountTransactionsController.TRANSACTION_PAGE_LIMIT == 0
```